cegs_portal/search/views/v1/experiment.py

```diff
@@ -31,12 +32,42 @@ def is_archived(self):
         except ObjectNotFoundError as e:
             raise Http404(str(e))
 
+    def request_options(self, request):
+        """
+        GET queries used:
+            analysis (single)
+                * Should be a valid analysis accession ID
+        """
+        options = super().request_options(request)
+        options["analysis"] = request.GET.get("analysis", None)
+        if options["analysis"] is not None:
+            validate_analysis_accession_id(options["analysis"])
+
+        return options
+
     def get(self, request, options, data, exp_id):
+        analyses = list(ExperimentSearch.all_analysis_id_search(exp_id))
+        analyses_list = []
+
+        if options["analysis"] is not None:
+            analysis_selected = options["analysis"]
+        else:
+            analysis_selected = ExperimentSearch.default_analysis_id_search(exp_id)
+
+        for analysis in analyses:
+            if analysis == options["analysis"]:
+                analyses_list.append(("selected", analysis))
+            else:
+                analyses_list.append(("", analysis))
+
         return super().get(
             request,
             options,
             {
                 "logged_in": not request.user.is_anonymous,
+                "analyses": analyses,
+                "analyses_list": analyses_list,
```

Comment:
You don't need to pass both `analyses` and `analyses_list` to the template; only the `analyses_list` data is used.

```suggestion
                "analyses": analyses_list,
```

and then update the template to use `analyses` instead of `analyses_list`.